src/ShareButton.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import React from 'react';
+import {Button} from 'react-bootstrap';
+import { injectIntl } from 'react-intl';
+import type { IntlShape } from 'react-intl';
+
+import ShareCompleteModal from './ShareCompleteModal';
+
+import './ShareButton.css';
+
+type ShareButtonProps = {
+    intl: IntlShape,
+    disabled: boolean,
+    onShowModal?: Function
+};
+
+type ShareButtonState = {
+    showShareComplete: boolean
+}
+
+class ShareButton extends React.Component<ShareButtonProps, ShareButtonState> {
+    static defaultProps = {
+        disabled: false,
+        showShareComplete: false
```

Comment:
`showShareComplete` is a state value, rather than props.

Reply:
Fixed.